File: lecture_010(RecursionwithArraylists)/l_009_printmazepath.java
import java.util.Scanner;
public class l_009_printmazepath {
    public static Scanner scn = new Scanner(System.in);
    public static void main(String[] args) {
        int n = scn.nextInt();
        int m = scn.nextInt();
        printMazePath2(1,1,n,m,"");
    }
    public static void printMazePath1(int sr, int sc, int dr, int dc, String path) {
        if(sr>dc || sc>dc) {
            return;
        }
        if(sr==dr && sc==dc) {
            System.out.println(path);
            return;
        }
        printMazePath1(sr,sc+1,dr,dc,path+"h");
        printMazePath1(sr+1,sc,dr,dc,path+"v");    
    }
    public static void printMazePath2(int sr, int sc, int dr, int dc, String path) {
        if(sr==dr && sc==dc) {
            System.out.println(path);
            return;
        }
        if(sc<dc) {
            printMazePath2(sr,sc+1,dr,dc,path+"h");
        }
        if(sr<dr) {
            printMazePath2(sr+1,sc,dr,dc,path+"v");
        }
    }
}

File: lecture_010(RecursionwithArraylists)/l_009_printmazepath.cpp
#include<iostream>
#include<vector>
#include<string>
using namespace std;

void printmazepath1(int sr, int sc, int dr, int dc, string path) {
    if(sr>dc || sc>dc) {
        return;
    }
    if(sr==dr && sc==dc) {
        cout<<path<<endl;
        return;
    }
    printmazepath1(sr,sc+1,dr,dc,path+"h");
    printmazepath1(sr+1,sc,dr,dc,path+"v");    
}
void printmazepath2(int sr, int sc, int dr, int dc, string path) {
    if(sr==dr && sc==dc) {
        cout<<path<<endl;
        return;
    }
    if(sc<dc) {
        printmazepath2(sr,sc+1,dr,dc,path+"h");
    }
    if(sr<dr) {
        printmazepath2(sr+1,sc,dr,dc,path+"v");
    }
}
int main() {
    int n,m;
    cin>>n>>m;
    printmazepath2(1,1,n,m,"");
}